File: src/main/java/com/brunodles/googleimagesapi/PageFetcher.java
package com.brunodles.googleimagesapi;

/**
 * Fetch pages for a determined Url
 */
public interface PageFetcher {
    /**
     * Fetch the url and return it as a String
     *
     * @param url the wanted url
     * @return the page as String
     */
    String fetchPage(String url);
}

File: search/src/main/kotlin/com/brunodles/googlesearchapi/PageFetcher.kt
package com.brunodles.googlesearchapi

/**
 * Fetch pages for a determined Url
 */
interface PageFetcher {
    /**
     * Fetch the url and return it as a String
     *
     * @param url the wanted url
     * @return the page as String
     */
    fun fetchPage(url: String): String
}
